qa/tasks/cephadm.py

```diff
@@ -816,6 +819,31 @@ def ceph_bootstrap(ctx, config):
                     '*', '--mode', '0755'],
                    check_status=False)
 
+        # default logrotate config for cluster logs does not work for teuthology
+        # where debug logging tends to be enabled and massive log files that
+        # are multiple Gigabytes can be generate in tests that last an hour or two
+        teuth_cluster_logrotate_config = """# created by cephadm
+/var/log/ceph/{{ fsid }}/*.log {
+    rotate 100
+    size 2G
+    compress
+    sharedscripts
+    postrotate
+        killall -q -1 {{ targets|join(' ') }} || pkill -1 -x '{{ targets|join('|') }}' || true
+    endscript
+    missingok
+    notifempty
+    su root root
+}"""
+        bootstrap_remote.write_file(
+            path='/etc/ceph/logrotate.conf',
+            data=teuth_cluster_logrotate_config)
+        _shell(ctx, cluster_name, bootstrap_remote,
```

Comment:
two remaining issues here

1. The upgrade tests will be bootstrapping with a version that can't handle the `write-custom-logrotate` commands. We need a fall back that manually writes the files out for that case
2. Because we're relying on cephadm to write these logrotate files out in the background, there is a chance the `Rotater` class tries to rotate the logs before cephadm has gotten a chance to write the logrotate config (keep in mind we're currently starting the Rotater going before the other hosts have even been added to the cluster). We'll probably need to make the Rotater be able to handle some failures caused by the logrotate config not being present.